c_glib/arrow-glib/compute.cpp

```diff
@@ -360,6 +360,108 @@ garrow_count_options_new(void)
 }
 
 
+typedef struct GArrowFilterOptionsPrivate_ {
+  arrow::compute::FilterOptions options;
+} GArrowFilterOptionsPrivate;
+
+enum {
+  PROP_NULL_SELECTION_BEHAVIOR = 1,
+};
+
+G_DEFINE_TYPE_WITH_PRIVATE(GArrowFilterOptions,
+                           garrow_filter_options,
+                           G_TYPE_OBJECT)
+
+#define GARROW_FILTER_OPTIONS_GET_PRIVATE(object)        \
+  static_cast<GArrowFilterOptionsPrivate *>(             \
+    garrow_filter_options_get_instance_private(          \
+      GARROW_FILTER_OPTIONS(object)))
+
+static void
+garrow_filter_options_set_property(GObject *object,
+                                   guint prop_id,
+                                   const GValue *value,
+                                   GParamSpec *pspec)
+{
+  auto priv = GARROW_FILTER_OPTIONS_GET_PRIVATE(object);
+
+  switch (prop_id) {
+  case PROP_NULL_SELECTION_BEHAVIOR:
+    priv->options.null_selection_behavior =
+      static_cast<arrow::compute::FilterOptions::NullSelectionBehavior>(g_value_get_enum(value));
+    break;
+  default:
+    G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+    break;
+  }
+}
+
+static void
+garrow_filter_options_get_property(GObject *object,
+                                  guint prop_id,
+                                  GValue *value,
+                                  GParamSpec *pspec)
+{
+  auto priv = GARROW_FILTER_OPTIONS_GET_PRIVATE(object);
+
+  switch (prop_id) {
+  case PROP_NULL_SELECTION_BEHAVIOR:
+    g_value_set_enum(value, priv->options.null_selection_behavior);
+    break;
+  default:
+    G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+    break;
+  }
+}
+
+static void
+garrow_filter_options_init(GArrowFilterOptions *object)
+{
+}
```

Comment:
Could you initialize `priv->options` explicitly like https://github.com/apache/arrow/pull/6694/files#diff-8c5b1fb4f5cd383aa0cc821cfbf30e23R715-R716 and destruct `priv-options` explicitly in `_finalize()` like https://github.com/apache/arrow/pull/6694/files#diff-8c5b1fb4f5cd383aa0cc821cfbf30e23R664-R672 ?

Reply:
I've added explicit calls of constructors and destructors not only for `FilterOptions`, but also `CastOptions`, `CountOptions`, `CompareOptions`, and `TakeOptions`.

Could you check again?

Reply:
+1
Thanks!